packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -0,0 +1,329 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { authentication, FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace externalAppAuthentication {
+  /*********** BEGIN REQUEST TYPE ************/
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type OriginalRequestInfo = IQueryMessageExtensionRequest | IActionExecuteInvokeRequest;
```

Comment:
@erinha @jekloudaMSFT @lakhveerk  Wha t do you think of this `OriginalRequestInfo` should follow the same naming pattern, which is to be named as `IOriginalRequestInfo` as `export type IInvokeResponse = IQueryMessageExtensionResponse | IActionExecuteResponse;` because both of `IQueryMessageExtensionRequest` and `IActionExecuteInvokeRequest` starts with `I` as well.

Reply:
I agree, it would be more consistent if we used the `I` prefix here too